internal/controller/controllers/agentclassification_controller.go

```diff
@@ -0,0 +1,187 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	aiv1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+	logutil "github.com/openshift/assisted-service/pkg/log"
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	"github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	AgentClassificationFinalizer = "agentclassification." + aiv1beta1.Group
+	ClassificationLabelPrefix    = "agentclassification." + aiv1beta1.Group + "/"
+)
+
+// AgentClassificationReconciler reconciles a AgentClassification object
+type AgentClassificationReconciler struct {
+	client.Client
+	Log logrus.FieldLogger
+}
+
+//+kubebuilder:rbac:groups=agent-install.openshift.io,resources=agentclassifications,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=agent-install.openshift.io,resources=agentclassifications/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=agent-install.openshift.io,resources=agentclassifications/finalizers,verbs=update
+//+kubebuilder:rbac:groups=agent-install.openshift.io,resources=agents,verbs=get;list;watch
+
+func (r *AgentClassificationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithFields(
```

Comment:
suggesting to add a request id to the log, it can help with debug, similar to https://github.com/openshift/assisted-service/blob/f28bddc8d95a89a4804b77b362dda721e3e14214/internal/controller/controllers/clusterdeployments_controller.go#L108

Reply:
Thanks, added to both controllers in the PR